File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0;
        int l = 0, r = height.length - 1;
        
        while(l <= r) {
            int h = Math.min(height[l], height[r]);
            int w = r - l;
            res = Math.max(res, h * w);
            if(h == height[l]) {
                l++;
            } else {
                r--;
            }
        }
        return res;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        area = 0
        while l < r:
            minHeight = min(height[l], height[r])
            width = r - l
            area = max(area, width * minHeight)
            if height[l] <= height[r]:
                 l+= 1
            else:
                r -= 1
        return area